iceberg-sql-catalog/src/lib.rs

```diff
@@ -436,6 +436,7 @@ impl Catalog for SqlCatalog {
     async fn update_table(self: Arc<Self>, commit: CommitTable) -> Result<Table, IcebergError> {
         let identifier = commit.identifier;
         let Some(entry) = self.cache.read().unwrap().get(&identifier).cloned() else {
+            #[allow(clippy::if_same_then_else)]
```

Comment:
I think something is wrong here? we fail if the table metadata is not in cache. Is this cache really necessary? Checking against a relational db is cheap.